src/app/features/courses/components/courses-list/components/courses-search/courses-search.component.ts

```diff
@@ -1,21 +1,51 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  OnDestroy,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import { fromEvent, map, filter, debounceTime, Subject, takeUntil } from 'rxjs';
 
 @Component({
   selector: 'app-courses-search',
   templateUrl: './courses-search.component.html',
   styleUrls: ['./courses-search.component.scss'],
 })
-export class CoursesSearchComponent implements OnInit {
-  searchValue: string = '';
+export class CoursesSearchComponent implements AfterViewInit, OnDestroy {
+  private readonly destroy$ = new Subject<boolean>();
 
   @Output()
   searchValueChange = new EventEmitter<string>();
 
+  @ViewChild('searchInput')
+  searchInput!: ElementRef;
+
   constructor() {}
 
-  ngOnInit(): void {}
+  ngAfterViewInit(): void {
+    if (this.searchInput.nativeElement) {
+      fromEvent(this.searchInput.nativeElement, 'keyup')
+        .pipe(
+          map((e: any) => {
+            return (e.target as HTMLInputElement).value;
+          }),
+          filter((text) => text.length > 2 || text.length === 0),
+          debounceTime(1000),
+          takeUntil(this.destroy$)
+        )
+        .subscribe({
+          next: (searchValue) => {
```

Comment:
There is no need to wrap observable result to object, if you don't use other properties to handle, such as `error` and `complete`, simplify to one-liner. This shorten form means, that it will be `next` only: 
`.subscribe(value => this.searchValueChange.emit(value ));`